test/extended/operators/check_crd_status.go

```diff
@@ -0,0 +1,127 @@
+package operators
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"k8s.io/kube-openapi/pkg/util/sets"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// checkCRDs returns a list of names of CRDs that don't have a "status" in the CRD schema
+// and also a subresource.status defined.
+// For now, it ignores the ones that are currently failing.
+func checkCRDs(crdItemList []apiextensionsv1.CustomResourceDefinition) []string {
+
+	// These CRDs, at the time this test was written, do not have a "status" in the CRD schema
+	// and subresource.status.
+	// These can be skipped for now but we don't want the number to increase.
+	// These CRDs should be tidied up over time.
+	//
+	exceptionsList := sets.NewString(
+		"builds.config.openshift.io",
+		"clusternetworks.network.openshift.io",
+		"consoleclidownloads.console.openshift.io",
+		"consoleexternalloglinks.console.openshift.io",
+		"consolelinks.console.openshift.io",
+		"consolenotifications.console.openshift.io",
+		"consoleplugins.console.openshift.io",
+		"consolequickstarts.console.openshift.io",
+		"consoleyamlsamples.console.openshift.io",
+		"egressnetworkpolicies.network.openshift.io",
+		"hostsubnets.network.openshift.io",
+		"imagecontentpolicies.config.openshift.io",
+		"imagecontentsourcepolicies.operator.openshift.io",
+		"machineconfigs.machineconfiguration.openshift.io",
+		"netnamespaces.network.openshift.io",
+		"networks.config.openshift.io",
+		"networks.operator.openshift.io",
+		"operatorpkis.network.operator.openshift.io",
+		"profiles.tuned.openshift.io",
+		"rangeallocations.security.internal.openshift.io",
+		"rolebindingrestrictions.authorization.openshift.io",
```

Comment:
This does not have a status field.

Reply:
correct.  `rolebindingrestrictions.authorization.openshift.io`, `rangeallocations.security.internal.openshift.io` and 16 other CRDs don't have a status field in their openapiv3 schema.

My unit test shows this:

```
CRD builds.config.openshift.io has no 'status' element in its schema
CRD clusternetworks.network.openshift.io has no 'status' element in its schema
CRD consoleclidownloads.console.openshift.io has no 'status' element in its schema
CRD consoleexternalloglinks.console.openshift.io has no 'status' element in its schema
CRD consolelinks.console.openshift.io has no 'status' element in its schema
CRD consolenotifications.console.openshift.io has no 'status' element in its schema
CRD consoleplugins.console.openshift.io has no 'status' element in its schema
CRD consolequickstarts.console.openshift.io has no 'status' element in its schema
CRD consoleyamlsamples.console.openshift.io has no 'status' element in its schema
CRD egressnetworkpolicies.network.openshift.io has no 'status' element in its schema
CRD hostsubnets.network.openshift.io has no 'status' element in its schema
CRD imagecontentpolicies.config.openshift.io has no 'status' element in its schema
CRD imagecontentsourcepolicies.operator.openshift.io has no 'status' element in its schema
CRD machineconfigs.machineconfiguration.openshift.io has no 'status' element in its schema
CRD netnamespaces.network.openshift.io has no 'status' element in its schema
CRD rangeallocations.security.internal.openshift.io has no 'status' element in its schema
CRD rolebindingrestrictions.authorization.openshift.io has no 'status' element in its schema
CRD securitycontextconstraints.security.openshift.io has no 'status' element in its schema

CRD networks.config.openshift.io has a status in the schema but no subresource.status
CRD networks.operator.openshift.io has a status in the schema but no subresource.status
CRD operatorpkis.network.operator.openshift.io has a status in the schema but no subresource.status
CRD profiles.tuned.openshift.io has a status in the schema but no subresource.status
CRD tuneds.tuned.openshift.io has a status in the schema but no subresource.status
```

and those are in the exceptions list.